[v0.3.1] Tests fails to build with shared library

kImageAnnotator fails to build the tests when using the CMake options `-DBUILD_TESTS='ON'` and `-DBUILD_SHARED_LIBS='ON'`:

```
[ 51%] Linking CXX executable StickerPickerTest
/usr/bin/ld: libKIMAGEANNOTATOR_STATIC.a(CapsLockStatusChecker.cpp.o): undefined reference to symbol 'XOpenDisplay'
/usr/bin/ld: /usr/lib/libX11.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/StickerPickerTest.dir/build.make:156: tests/StickerPickerTest] Error 1
make[2]: Leaving directory '/storage/linux/abs/kimageannotator/src/build'
make[1]: *** [CMakeFiles/Makefile2:471: tests/CMakeFiles/StickerPickerTest.dir/all] Error 2
make[1]: Leaving directory '/storage/linux/abs/kimageannotator/src/build'
make: *** [Makefile:161: all] Error 2
make: Leaving directory '/storage/linux/abs/kimageannotator/src/build'
```

The tests builds fine with static library (using `-DBUILD_TESTS='ON'` and  `-DBUILD_SHARED_LIBS='OFF'`).

This issue is caused because `target_link_libraries` is being used with the `PRIVATE` keyword for linking the dependency libraries. According to the [CMake documentation](https://cmake.org/cmake/help/v3.17/command/target_link_libraries.html), the `PUBLIC` keyword should be used for the libraries to be part of the link interface.

The following changes fixes the build for me:

```patch
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,10 +67,10 @@ target_include_directories(kImageAnnotat
 						   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
 						   )
 
-target_link_libraries(kImageAnnotator PUBLIC Qt5::Widgets Qt5::Svg PRIVATE kColorPicker::kColorPicker)
+target_link_libraries(kImageAnnotator PUBLIC Qt5::Widgets Qt5::Svg kColorPicker::kColorPicker)
 
 if (UNIX)
-	target_link_libraries(kImageAnnotator PRIVATE X11)
+	target_link_libraries(kImageAnnotator PUBLIC X11)
 endif ()
 
 target_compile_definitions(kImageAnnotator PRIVATE KIMAGEANNOTATOR_LIB)

```

If these changes are acceptable I can open a pull request.

System Information:
- OS: Arch Linux x86_64
- CMake: 3.17.3
- Compiler: gcc 10.1.0
- Qt: 5.15.0
- kColorPicker: 0.1.4

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,10 +67,10 @@ target_include_directories(kImageAnnotator
 						   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
 						   )
 
-target_link_libraries(kImageAnnotator PUBLIC Qt5::Widgets Qt5::Svg PRIVATE kColorPicker::kColorPicker)
+target_link_libraries(kImageAnnotator PUBLIC Qt5::Widgets Qt5::Svg kColorPicker::kColorPicker)
 
 if (UNIX)
-	target_link_libraries(kImageAnnotator PRIVATE X11)
+	target_link_libraries(kImageAnnotator PUBLIC X11)
 endif ()
 
 target_compile_definitions(kImageAnnotator PRIVATE KIMAGEANNOTATOR_LIB)

